### PR TITLE
chore: Move schema source/generation out of samtranslator package

### PR DESCRIPTION
**Pull request by aahung**
### Description of changes

The generated schema files stay in `samtranslator/schema`, everything else moved to `schema_source`.
This removes `pydantic` from our pypi requirement

### Checklist

- [ ] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [ ] Using correct values
    - [ ] Using wrong values
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -51,21 +51,21 @@ update-schema-data:
...
 	# Update CloudFormation schema
 	curl -o .tmp/cloudformation.schema.json https://raw.githubusercontent.com/awslabs/goformation/master/schema/cloudformation.schema.json
 
 	# Add CloudFormation docs to CloudFormation schema
-	python bin/add_docs_cfn_schema.py --schema .tmp/cloudformation.schema.json --docs samtranslator/schema/cloudformation-docs.json > samtranslator/schema/cloudformation.schema.json
+	python bin/add_docs_cfn_schema.py --schema .tmp/cloudformation.schema.json --docs schema_source/cloudformation-docs.json > schema_source/cloudformation.schema.json
 
 schema:
-	python samtranslator/schema/schema.py --sam-schema samtranslator/schema/sam.schema.json --cfn-schema samtranslator/schema/cloudformation.schema.json --unified-schema samtranslator/schema/schema.json
+	python -m schema_source.schema --sam-schema samtranslator/schema/sam.schema.json --cfn-schema schema_source/cloudformation.schema.json --unified-schema samtranslator/schema/schema.json
```
Comment (hoffa) [inline Makefile:68]: We can move `sam.schema.json` as well, it's used to generate the final schema.

Comment (aahung) [reply Makefile]: done

File: Makefile [hunk 2]
```diff
@@ -26,8 +26,8 @@ black:
 black-check:
 	# Checking latest schema was generated (run `make schema` if this fails)
 	mkdir -p .tmp
```
Comment (hoffa) [inline Makefile:28]: Need to change `MANIFEST.in` as well.

**Review by hoffa — APPROVED** [review]

**Review by ssenchenko — APPROVED** [review]

**Comment by musicinmybrain** [comment]
As a result of this PR, `schema_source` is installed in `site-packages` beginning with 1.60.0. I expect this was not intentional. This directory could be excluded in:

https://github.com/aws/serverless-application-model/blob/480be09b609e7fad5d27d77d420af97489b32096/setup.py#L62

**Comment by aahung** [comment]
@musicinmybrain https://github.com/aws/serverless-application-model/releases/tag/v1.62.0 is out

**Pull request merged**